pyccel/codegen/printing/pycode.py

```diff
@@ -124,13 +124,20 @@ def _print_FunctionDef(self, expr):
         return code
 
     def _print_Return(self, expr):
-        code = ''
+        rhs_list = []
+        lhs_list = []
+        expr_return_vars = []
         if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
-        if expr.expr:
-            ret = ','.join([self._print(i) for i in expr.expr])
-            code += 'return {}'.format(ret)
-        return code
+            for i in expr.stmt.body:
+                rhs_list.append(i.rhs)
+                lhs_list.append(i.lhs)
+        for a in expr.expr:
+            if lhs_list and a not in lhs_list:
+                expr_return_vars.append(a)
```

Comment:
Can you give an example of where this block is needed?

Reply:
we don't want to print Dummy variables from `expr.expr`, in all cases `expr.expr` will contain either dummy variables or user's variables, that's why i collect unneeded dummy variables from `expr.stmt` if they exist and avoid them is this `if` block, the other variables that come from `rhs`s 
in`expr.stmt`  I'm printing them in the return.

Reply:
But can we ever have variables in `expr.expr` which don't exist in `expr.stmt` if there is an `expr.stmt`?

Reply:
yes, if you have them as arguments

Reply:
Ah ok :+1: